br/pkg/restore/snap_client/import.go

```diff
@@ -646,6 +683,120 @@ func (importer *SnapFileImporter) buildDownloadRequest(
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			if req == nil {
+				continue
+			}
+			sstMeta.ApiVersion = apiVersion
+			cfReq, exists := downloadReqMap[file.Cf]
+			if !exists {
+				cfReq = &import_sstpb.DownloadRequest{
+					Sst:            req.Sst,
+					Name:           req.Name,
+					RewriteRule:    req.RewriteRule,
+					StorageBackend: req.StorageBackend,
+					CipherInfo:     req.CipherInfo,
+					StorageCacheId: req.StorageCacheId,
+					RequestType:    req.RequestType,
+					Context:        req.Context,
+					Ssts:           make(map[string]*import_sstpb.SSTMeta),
+				}
+				downloadReqMap[file.Cf] = cfReq
+			}
+			// check the rewrite rule the same
+			if !(bytes.Equal(cfReq.RewriteRule.NewKeyPrefix, req.RewriteRule.NewKeyPrefix) &&
+				bytes.Equal(cfReq.RewriteRule.OldKeyPrefix, req.RewriteRule.OldKeyPrefix) &&
+				cfReq.RewriteRule.IgnoreAfterTimestamp == req.RewriteRule.IgnoreAfterTimestamp &&
+				cfReq.RewriteRule.IgnoreBeforeTimestamp == req.RewriteRule.IgnoreBeforeTimestamp) {
+				log.Panic("rewrite rule mismatch", zap.Reflect("cfReq", cfReq.RewriteRule), zap.Reflect("req", req.RewriteRule))
+			}
+			cfReq.Ssts[req.Name] = &sstMeta
+		}
+		// fallback to single download if there is only one sst
+		for _, req := range downloadReqMap {
+			if len(req.Ssts) <= 1 {
+				req.Ssts = nil
+			}
+		}
+		if len(downloadReqMap) > 0 {
+			downloadReqs = append(downloadReqs, downloadReqMap)
+		}
+	}
+
+	eg, ectx := errgroup.WithContext(ctx)
+	for _, p := range regionInfo.Region.GetPeers() {
+		peer := p
+		eg.Go(func() error {
+			tokenCh := importer.downloadTokensMap.acquireTokenCh(peer.GetStoreId(), importer.concurrencyPerStore)
+			select {
+			case <-ectx.Done():
+				return ectx.Err()
+			case <-tokenCh:
+			}
+			defer func() {
+				importer.releaseToken(tokenCh)
+			}()
+			for _, downloadReqMap := range downloadReqs {
+				for _, req := range downloadReqMap {
+					var err error
+					var resp *import_sstpb.DownloadResponse
+					resp, err = utils.WithRetryV2(ectx, utils.NewDownloadSSTBackoffStrategy(), func(ctx context.Context) (*import_sstpb.DownloadResponse, error) {
+						dctx, cancel := context.WithTimeout(ctx, gRPCTimeOut)
+						defer cancel()
+						if len(req.Ssts) == 0 {
+							// fallback to single download
+							return importer.importClient.DownloadSST(dctx, peer.GetStoreId(), req)
+						} else {
+							return importer.importClient.BatchDownloadSST(dctx, peer.GetStoreId(), req)
+						}
+					})
+					if err != nil {
+						return errors.Trace(err)
+					}
+					if resp.GetError() != nil {
+						return errors.Annotate(berrors.ErrKVDownloadFailed, resp.GetError().GetMessage())
+					}
+					if resp.GetIsEmpty() {
+						log.Warn("download file skipped", zap.String("filename", req.Name),
+							logutil.Region(regionInfo.Region), zap.Error(berrors.ErrKVRangeIsEmpty))
+						continue
+					}
+
+					mu.Lock()
+					sstMeta := req.Sst
```

Comment:
Need handle `req.Ssts` too?

Reply:
For TiKV server, the input is `req.Ssts` and the output target is `req.Sst`. Therefore, just get the `req.Sst` to obtain the output of merged SST file.